src/js/lib/monster.util.js

```diff
@@ -1245,6 +1234,48 @@ define(function(require) {
 			);
 	}
 
+	/**
+	 * Generates a string of `length` random characters chosen from either a
+	 * preset or a custom string of characters.
+	 * @param  {Number} length  Number of characters to include.
+	 * @param  {String} pPreset Characters to choose from.
+	 * @return {String}         A string of random characters.
+	 */
+	function randomString(length, pPreset) {
+		if (!_.isNumber(length)) {
+			throw new TypeError('Expected a number for `length`');
+		}
+		if (!_.isUndefined(pPreset) && !_.isString(pPreset)) {
```

Comment:
What should happen if `pPreset` is an empty string? Currently `monster.util.randomString(5, '')` returns an empty string.

Reply:
Good catch!

In my opinion, that's the expected behavior of the method, as it does what it's told to do, namely, pick five characters out of an empty set of characters.

One improvement from this would be to check if the `preset` is empty and return an empty string right away instead of starting the chaining operation.